$NetBSD: patch-utilities_CMakeLists.txt,v 1.1 2023/05/23 21:01:12 adam Exp $

Split configuration between mysql-client and mysql-server.
Build comp_err also in a cross build that has an emulator to run it
(CMAKE_CROSSCOMPILING_EMULATOR, the IRIX irix-exec bridge).
IRIX has neither ppoll nor kqueue: poll() in vio, as on Windows.
The bundled boost's MIPSpro workaround is not for clang, which defines
_COMPILER_VERSION on IRIX too.

--- utilities/CMakeLists.txt.orig
+++ utilities/CMakeLists.txt
@@ -25,11 +25,18 @@
 DISABLE_MISSING_PROFILE_WARNING()
 MSVC_CPPCHECK_DISABLE()
 
-IF(NOT CMAKE_CROSSCOMPILING)
+IF(WITHOUT_SERVER)
+  SET(CLIENT_SKIP_INSTALL SKIP_INSTALL)
+ELSE()
+  SET(CLIENT_SKIP_INSTALL "")
+ENDIF()
+
+IF(NOT CMAKE_CROSSCOMPILING OR CMAKE_CROSSCOMPILING_EMULATOR)
   MYSQL_ADD_EXECUTABLE(comp_err
     comp_err.cc
     COMPONENT Test
     LINK_LIBRARIES mysys
+    ${CLIENT_SKIP_INSTALL}
     )
 ENDIF()
 
@@ -118,12 +125,14 @@
   my_print_defaults.cc
   COMPONENT Server
   LINK_LIBRARIES mysys
+  ${CLIENT_SKIP_INSTALL}
   )
 MYSQL_ADD_EXECUTABLE(perror
   perror.cc
   COMPONENT Server
   DEPENDENCIES GenError
   LINK_LIBRARIES mysys
+  ${CLIENT_SKIP_INSTALL}
   )
 IF(WITH_LZ4 STREQUAL "bundled")
   MYSQL_ADD_EXECUTABLE(lz4_decompress
@@ -137,6 +146,7 @@
   zlib_decompress.cc
   COMPONENT Server
   LINK_LIBRARIES ext::zlib mysys
+  ${CLIENT_SKIP_INSTALL}
   )
 
 # All targets below belong to COMPONENT Server and depend on InnoDB.
